Sqrt binary search

Shashwat79802 opened the pull request:
### Added the precision method in the program.

Made the following changes to the code :
1. Added a method to calculate the decimal value of the square root of a number that is not a perfect square.
2. Modified the binary search method.
3. Edited some documentation.

Earlier the output was : 
![earlier](https://user-images.githubusercontent.com/63868528/185152474-588281c0-9f41-4736-a9cf-266d3d5f171b.jpg)

Now the output is : 
![now](https://user-images.githubusercontent.com/63868528/185152481-e790de30-3b2d-4a45-8fc8-37d81be08683.jpg)

siriak reviewed (changes requested):
Please only include relevant changes in this PR. src/main/java/com/thealgorithms/others/SieveOfEratosthenes.java changes seem to be unrelated

Shashwat79802 commented:
So shall I withdraw my PR? 'Cause I felt that I should add the precision method to have a precise square root and according to it, I made changed in the code.

siriak commented:
What does src/main/java/com/thealgorithms/others/SieveOfEratosthenes.java have to do with square root?

Shashwat79802 commented:
I don't know what went wrong with my PR, but the precision thing that I added is for the program to calculate **square root using Binary Search**.

And the program for the Sieve of Eratosthenes, I optimized by removing some data structures taking extra space, and some unnecessary loops.

PR - #3220 is for the Sieve of Eratosthenes and in that, I have explained everything that I did.
and, 
PR - #3221 is for the sqrtBinarySearch in which I added the code to calculate the precision of the square root also.

Shall I make the PR's again if they are messed up?

siriak commented on src/main/java/com/thealgorithms/searches/SquareRootBinarySearch.java:
```diff
@@ -2,31 +2,35 @@
 
+    public SquareRootBinarySearch() {
+    }
+
     /**
      * This is the driver method.
      *
      * @param args Command line arguments
      */
     public static void main(String args[]) {
         Scanner sc = new Scanner(System.in);
+
         System.out.print("Enter a number you want to calculate square root of : ");
         int num = sc.nextInt();
-        long ans = squareRoot(num);
-        System.out.println("The square root is : " + ans);
+
+        int precision = 5;
+
+        double ans = squareRoot(num, precision);
+        System.out.println("The square root of " + num + " is : " + ans);
     }
```
Please use JUnit tests instead

siriak commented on src/main/java/com/thealgorithms/searches/SquareRootBinarySearch.java:
```diff
@@ -37,24 +41,48 @@ public static void main(String args[]) {
      * @return answer
      */
-    private static long squareRoot(long num) {
+    private static double squareRoot(long num, int precision) {
```
It's not precision, but rather number of iterations

siriak commented on src/main/java/com/thealgorithms/searches/SquareRootBinarySearch.java:
```diff
@@ -37,24 +41,48 @@ public static void main(String args[]) {
+            }
+
+            else if (mid * mid >= num)
                 r = mid - 1;
+
+            else
```
Please remove redundant empty lines

Shashwat79802 replied:
Done

siriak commented on src/main/java/com/thealgorithms/searches/SquareRootBinarySearch.java:
```diff
@@ -37,24 +41,48 @@ public static void main(String args[]) {
+     * of the root as per the provided precision value.
+     */
+
+    private static double decimal(long n, double root, int precision) {
```
Why is it called `decimal`?

Shashwat79802 replied:
Because it's giving the decimal value till the provided precision value. Shall I change it to something else?

siriak commented on src/main/java/com/thealgorithms/others/SieveOfEratosthenes.java:
```diff
@@ -32,49 +30,39 @@ public class SieveOfEratosthenes {
-        // Create array where index is number and value is flag - is that number a prime or not.
-        // size of array is n + 1 cause in Java array indexes starts with 0
-        Type[] numbers = new Type[n + 1];
+    public static void findPrimesTill(int n, boolean[] notPrime) {
+        
+        //Initializing indexes 0 & 1 to be true as they are neither prime nor composite.
+        notPrime[0] = true;
+        notPrime[1] = true;
```
You have made this branch from your previous branch, not main

siriak replied:
That's why your other changes are included

Shashwat79802 replied:
If the changes seems to be relevant, shall I make a new PR with a fresh branch?

siriak reviewed (commented):
This bit with adding more decimal places to the root is useful in certain circumstances, but it complicates the algorithm and doesn't use binary search. So, I would rather not include it here.

siriak commented:
Thanks for suggesting this improvement!
